calour/dsfdr.py

```diff
@@ -114,106 +123,115 @@ def dsfdr(data, labels, method='meandiff', transform='rankdata', alpha=0.1, nump
         for cperm in range(numperm):
             np.random.shuffle(labels)
             p[labels == 0, cperm] = k1
-        p2 = np.ones(p.shape) * k2
-        p2[p > 0] = 0
+         p2 = np.ones(p.shape) * k2
+         p2[p > 0] = 0
         mean1 = np.dot(data, p)
         mean2 = np.dot(data, p2)
         u = np.abs(mean1 - mean2)
 
-    elif method == 'mannwhitney':
-        method = mannwhitney
-        tstat = method(data, labels)
-        t = np.abs(tstat)
-        u = np.zeros([numbact, numperm])
-        for cperm in range(numperm):
-            rlabels = np.random.permutation(labels)
-            rt = method(data, rlabels)
-            u[:, cperm] = rt
-
-    elif method == 'kruwallis':
-        method = kruwallis
-        tstat = method(data, labels)
-        t = np.abs(tstat)
-        u = np.zeros([numbact, numperm])
+    elif method == 'mannwhitney' or method == 'kruwallis' or method == 'stdmeandiff':
+        if method == 'mannwhitney':
+            method = mannwhitney
+        if method == 'kruwallis':
+            method = kruwallis
+        if method == 'stdmeandiff':
+            method = stdmeandiff
+            
+        tstat=method(data,labels)
+        t=np.abs(tstat)
+        u=np.zeros([numbact,numperm])
         for cperm in range(numperm):
-            rlabels = np.random.permutation(labels)
-            rt = method(data, rlabels)
-            u[:, cperm] = rt
-
-    elif method == 'stdmeandiff':
-        method = stdmeandiff
-        tstat = method(data, labels)
-        t = np.abs(tstat)
-        u = np.zeros([numbact, numperm])
-        for cperm in range(numperm):
-            rlabels = np.random.permutation(labels)
-            rt = method(data, rlabels)
-            u[:, cperm] = rt
+            rlabels=np.random.permutation(labels)
+            rt=method(data,rlabels)
+            u[:,cperm]=rt
 
     elif method == 'spearman' or method == 'pearson':
-        # do fast matrix multiplication based correlation
+        # fast matrix multiplication based correlation
         if method == 'spearman':
             data = rankdata(data)
             labels = sp.stats.rankdata(labels)
-        meanval = np.mean(data, axis=1).reshape([data.shape[0], 1])
-        data = data - np.repeat(meanval, data.shape[1], axis=1)
-        labels = labels - np.mean(labels)
-        tstat = np.dot(data, labels)
-        t = np.abs(tstat)
+        meanval=np.mean(data,axis=1).reshape([data.shape[0],1])
+        data=data-np.repeat(meanval,data.shape[1],axis=1)
+        labels=labels-np.mean(labels)
+        tstat=np.dot(data, labels)
+        t=np.abs(tstat)
         permlabels = np.zeros([len(labels), numperm])
         for cperm in range(numperm):
-            rlabels = np.random.permutation(labels)
-            permlabels[:, cperm] = rlabels
-        u = np.abs(np.dot(data, permlabels))
+            rlabels=np.random.permutation(labels)
+            permlabels[:,cperm] = rlabels
+        u=np.abs(np.dot(data,permlabels))
```

Comment:
same problem lines 226-234
missing space before and after '=' (pep8 needs it otherwise it complains...)